crates/services/producer/src/block_producer.rs

```diff
@@ -382,6 +384,27 @@ where
     }
 }
 
+impl<ViewProvider, TxPool, Executor, GasPriceProvider, ConsensusProvider>
+    Producer<ViewProvider, TxPool, Executor, GasPriceProvider, ConsensusProvider>
+where
+    ViewProvider: HistoricalView + 'static,
+    ViewProvider::LatestView: BlockProducerDatabase,
+    Executor: ports::StorageReadReplayRecorder + 'static,
+    GasPriceProvider: GasPriceProviderConstraint,
+    ConsensusProvider: ConsensusParametersProvider,
+{
+    /// Re-executes an old block, getting the storage read events.
+    pub async fn storage_read_replay(
```

Comment:
Why `storage_read_replay` is an `async` method? It seems all operations are sync inside.

Reply:
Now needed since I use spawn_fifo like dry_run does